modern/src/maki-interpreter/interpreter.js

```diff
@@ -295,7 +312,8 @@ export function* interpret(start, program, stack = []) {
       case 96: {
         const classesOffset = command.arg;
         const Klass = classes[classesOffset];
-        const klassInst = new Klass();
+        const system = variables[0].getValue();
+        const klassInst = new Klass(null, system.scriptGroup);
```

Comment:
We are kind of abusing the MAKI object notion of a parent to get a reference to the scriptGroup in Maki Objects incase they need to do something related to the MAKI tree they have a reference

Reply:
Seems like a reasonable hack. I wonder if we can find an experiment that will tell us what the parent of a created object is.

Reply:
One thing that's a bit odd here (and I guess is intentional) is that we end up with a node that has another node as its parent but that parent does not have the new node as a child.

All of this will get reconsidered when (if?) we switch to Redux, so I'm okay leaving it as is for now.

Reply:
> Seems like a reasonable hack. I wonder if we can find an experiment that will tell us what the parent of a created object is.

Yea that would be great. `PopupMenu` is really bizarre because it never actually gets added to any object in the MAKI script, it just gets created and then later `popAtMouse` runs, so it really doesn't seem to have any parent.

And yea, the tree structure gets kinda weird. hopefully we'll have better ideas of how to do it in the future :D